### PR TITLE
feat(@cubejs-backend/query-orchestrator): set pool options from server config

### PR DESCRIPTION
**Pull request by artem-khromov**
Allows to set Redis pool options from server config.
Can be used to pick another Redis client or define properties (ex. poolMax) without environment variables.

**Check List**
- [x] Tests has been run in packages where changes made if available
- [x] Linter has been run for changed code
- [x] Tests for the changes have been added if not covered yet
- [ ] Docs have been added / updated if required

**Issue Reference this PR resolves**

None

**Description of Changes Made (if issue reference is not provided)**

Added `poolOptions` to `OrchestratorOptions` interface
`poolOptions` are now passed to `RedisPool` constructor

### CODE REVIEW

File: packages/cubejs-server-core/core/index.d.ts [hunk 1]
```diff
@@ -44,6 +44,12 @@ export interface CreateOptions {
 
 export interface OrchestratorOptions {
   redisPrefix?: string;
+  poolOptions?: {
+    poolMin?: number;
+    poolMax?: number;
+    createClient?: (redisUrl: string) => {}
+    destroyClient?: (client) => {}
```
Comment (hassankhan) [inline packages/cubejs-server-core/core/index.d.ts:51]: I think we should re-use the types from `redis` here:

```suggestion
    createClient?: (redisUrl: string) => RedisClient
    destroyClient?: (client: RedisClient) => void
```

**Review by hassankhan — CHANGES_REQUESTED** [review]

**Comment by hassankhan** [comment]
Thanks for the PR @artem-khromov 🎉 Added a comment to the PR, let us know if you have any problems 👍

File: packages/cubejs-server-core/core/index.d.ts [hunk 2]
```diff
@@ -44,6 +44,12 @@ export interface CreateOptions {
 
 export interface OrchestratorOptions {
   redisPrefix?: string;
+  poolOptions?: {
+    poolMin?: number;
+    poolMax?: number;
+    createClient?: (redisUrl: string) => RedisClient
+    destroyClient?: (client: RedisClient) => void
```
Comment (ovr) [inline packages/cubejs-server-core/core/index.d.ts:51]: `RedisClient` is not defined type here.

**Comment by ovr** [comment]
@artem-khromov 

I cherry-picked your commit and improve typings support by re-exporting it from RedisPool inside https://github.com/cube-js/cube.js/pull/1596.

Thank you for your contribution 🍻

**Pull request closed without merging**